utils/gov/formatData.ts

```diff
@@ -0,0 +1,177 @@
+        case 'PROPOSAL_STATUS_VOTING_PERIOD':
+            return 'ACTIVE';
+        case 'PROPOSAL_STATUS_UNSPECIFIED':
+            return undefined;
+        default:
+            return undefined;
+        
+    }
+}
+  
+export function formatProposalType(type_url: string){
+    return (proposalTypes as any)[type_url] || 'Proposal type not found';
+}
+interface FinalTallyResult {
+    yes: string;
+    abstain: string;
+    no: string;
+    no_with_veto: string;
+  }
+  
+  export interface VoteData {
+    YesAmount: string,
+    NoAmount: string,
+    AbstainAmount: string,
+    VetoAmount: string,
+    Yes: string;
+    Abstain: string;
+    No: string;
+    Veto: string;
+  }
+
+export function calculateVotePercentages(finalTallyResult: FinalTallyResult): VoteData {
+  // Destructure the vote counts from the object
+  const { yes, abstain, no, no_with_veto } = finalTallyResult;
+
+  const factor = BigInt(1e18);
+  // Convert the vote counts to BigInts
+  const yesVotes = BigInt(yes);
+  const abstainVotes = BigInt(abstain);
+  const noVotes = BigInt(no);
+  const noWithVetoVotes = BigInt(no_with_veto);
+
+  const yesNumber = Number(yesVotes / factor) + Number(yesVotes % factor) / Number(factor);
+  const abstainNumber = Number(abstainVotes / factor) + Number(abstainVotes % factor) / Number(factor);
+  const noNumber = Number(noVotes / factor) + Number(noVotes % factor) / Number(factor);
+  const noWithVetoNumber = Number(noWithVetoVotes / factor) + Number(noWithVetoVotes % factor) / Number(factor);
+
+  //console.log(yesNumber);
+  const totalVotesBigInt = yesVotes+noVotes+abstainVotes+noWithVetoVotes;
+  const totalVotes = yesNumber + abstainNumber + noNumber + noWithVetoNumber;
+
+  if(totalVotesBigInt==0n){
+    return {
+        YesAmount: "0",
+        NoAmount: "0",
+        AbstainAmount: "0",
+        VetoAmount: "0",
+        Yes: "0",
+        Abstain: "0",
+        No: "0",
+        Veto: "0",
+      };
+  }
+  
+  // Calculate percentages and format them as strings
+  const yesPercentage = (Number(yesVotes*BigInt(10000) / totalVotesBigInt )/100).toFixed(2);
+  const abstainPercentage = (Number(abstainVotes*BigInt(1000000) / totalVotesBigInt )/10000).toFixed(2);
+  const noPercentage = (Number(noVotes*BigInt(10000) / totalVotesBigInt )/100).toFixed(2);
+  const noWithVetoPercentage = (Number(noWithVetoVotes*BigInt(10000) / totalVotesBigInt )/100).toFixed(2);
+
+  // Convert the vote counts to strings with two decimal places
+  const yesString = yesNumber.toFixed(2);
+  const abstainString = abstainNumber.toFixed(2);
+  const noString = noNumber.toFixed(2);
+  const noWithVetoString = noWithVetoNumber.toFixed(2);
+
+  return {
+    YesAmount: yesString,
+    NoAmount: noString,
+    AbstainAmount: abstainString,
+    VetoAmount: noWithVetoString,
+    Yes: yesPercentage,
+    Abstain: abstainPercentage,
+    No: noPercentage,
+    Veto: noWithVetoPercentage,
+  };
+}
+
+export function formatDeposit(amount: string): string{
```

Comment:
we have formatting for this. Let's just use displayBalance function from our formatting library

Reply:
Done this. Using displayAmount() now